src/main/java/com/avispl/dal/communicator/cisco/CiscoCommunicator.java

```diff
@@ -461,8 +461,9 @@ public String dial(DialDevice dialDevice) throws Exception {
             }
             Thread.sleep(1000);
         }
-        throw new RuntimeException(String.format("An error occurred during dialing out to %s with protocol %s.",
-                dialString, dialDevice.getProtocol()));
+        // Some devices (like SX20) appear slower than the other ones, so if we throw an exception here - it would
+        // be populate to UI, but call status will still be updated afterwards.
+        return null;
```

Comment:
i dont think this is the right approach.  I see several approaches

1. use a standardized exception so that this specific situation can be handled properly by all call flows that utilize it.  
option 1 is probably the easiest and best option, however there still will need changes to cp

2. make this call asynchronous
I suppose that there will likely be alot of code adjustment here to support this (especially in cp)

3. I suppose this code will work but then we need to handle null properly in cp (and other layers).  but then we need logging at least, imo.  and idk what other devs might think about just returning null,  i feel like vlad or shervin might not be so happy with merely returning null

Reply:
Agree, however I have seen this in a couple of other communicators out there, like CiscoX, PolycomVSX and others

Reply:
This is how other communicators work, and CP is handling this case, and logging it.